pandas-stubs/core/frame.pyi

```diff
@@ -268,32 +268,46 @@ class DataFrame(NDFrame, OpsMixin):
     @overload
     def to_dict(
         self,
-        orient: Literal["dict", "list", "series", "split", "tight", "index"],
+        orient: Literal["dict", "list", "series", "index"],
         into: Mapping | type[Mapping],
+        index: Literal[True] = True,
     ) -> Mapping[Hashable, Any]: ...
     @overload
     def to_dict(
         self,
-        orient: Literal["dict", "list", "series", "split", "tight", "index"] = ...,
+        orient: Literal["dict", "list", "series", "index"] = ...,
         *,
         into: Mapping | type[Mapping],
+        index: Literal[True] = True,
     ) -> Mapping[Hashable, Any]: ...
     @overload
     def to_dict(
         self,
-        orient: Literal["dict", "list", "series", "split", "tight", "index"] = ...,
+        orient: Literal["dict", "list", "series", "index"] = ...,
         into: None = ...,
+        index: Literal[True] = True,
     ) -> dict[Hashable, Any]: ...
     @overload
     def to_dict(
         self,
         orient: Literal["records"],
         into: Mapping | type[Mapping],
+        index: Literal[True] = True,
     ) -> list[Mapping[Hashable, Any]]: ...
     @overload
     def to_dict(
-        self, orient: Literal["records"], into: None = ...
+        self,
+        orient: Literal["records"],
+        into: None,
+        index: Literal[True] = True,
     ) -> list[dict[Hashable, Any]]: ...
+    @overload
+    def to_dict(
+        self,
+        orient: Literal["split", "tight"] = ...,
+        into: None = ...,
+        index: bool = ...,
+    ) -> dict[Hashable, Any]: ...
```

Comment:
You'll have to copy the previous overloads for the different parameter combinations that were there before for `orient: Literal["split", "tight"]` when they were part of the bigger set of values.

Reply:
Not quite undertood this are you saying that previous overloads with all the values of `orient` have to copied and we just have to change the index values to `bool` then

Reply:
No, we had previous overloads that had `orient: Literal["dict", "list", "series", "split", "tight", "index"]`.  Now for each of them, you will split them into two, one with `orient: Literal["dict", "list", "series", "index"]`, the other with `orient: Literal["split", "tight"]` .  The first one will allow `index: Literal[True]`.  The second one would allow `index: bool = ...`

Reply:
Actually I am not able to understand this mypy error as we have overload for `Literal["records"]` but it is not able to get it